src/commonHelper.ts

```diff
@@ -6,57 +6,122 @@
 import * as Constants from './constants';
-import { CommandObserver } from './CommandObserver';
+import { CommandObserver } from './commandObserver';
 import * as compareVersions from 'compare-versions';
-const regex = /(?<="Microsoft.DataTools.Schema.Tasks.PostgreSql.Sdk\/)(.*?)(?=")/;
+import * as nls from 'vscode-nls';
+
+const localize = nls.loadMessageBundle();
+const xml2js = require('xml2js');
+
+const sdkName: string = "Microsoft.DataTools.Schema.Tasks.PostgreSql.Sdk";
+const enum sdkReference {
+    Attribute = 1,
+    Element = 2
+};
+
+export type ProjectInformation = {
+    path: string;
+    sdkVersion: string;
+    sdkReferenceType: sdkReference;
+}
 
 export async function checkProjectVersion(minRequiredSDK: string, maxRequiredSDK: string, projects: string[], commandObserver: CommandObserver) : Promise<string[]> {
-	let unsupportedProjectsMap: string[] = [];
+	let unsupportedProjectsMap = new Array<ProjectInformation>();
     for (let project of projects) {
         let projectFileText = fs.readFileSync(project, 'utf8');
-            let sdkVersion;
-            if ((sdkVersion = regex.exec(projectFileText)) !== null) {
-				if (!(compareVersions.compare(sdkVersion[0], minRequiredSDK, '>=') && compareVersions.compare(sdkVersion[0], maxRequiredSDK, '<='))) {
-                    unsupportedProjectsMap.push(project);
-                }
+        let projectInfo = readXML(project, projectFileText, commandObserver);
+        if (projectInfo) {
+            if (!(compareVersions.compare(projectInfo.sdkVersion, minRequiredSDK, '>=') && compareVersions.compare(projectInfo.sdkVersion, maxRequiredSDK, '<='))) {
+                unsupportedProjectsMap.push(projectInfo);
             }
+        }
     }
 
     if (unsupportedProjectsMap.length > 0) {
 		promptToUpdateVersion(unsupportedProjectsMap, maxRequiredSDK, commandObserver);
 	}
 
-	return Promise.resolve(unsupportedProjectsMap);
+	return Promise.resolve(unsupportedProjectsMap.map(p => p.path));
+}
+
+function readXML(project: string, fileText: string, commandObserver: CommandObserver): ProjectInformation {
+    var parser = new xml2js.Parser();
+    let version: string;
+    let referenceType: sdkReference;
+    parser.parseString(fileText, function(err, result) {
+        if (result && result.Project && result.Project.$ && result.Project.$.Sdk) {
+            referenceType = sdkReference.Attribute;
+            let splitArray: string[] = result.Project.$.Sdk.split('/');
+            if (splitArray.length === 2 && splitArray[0].replace(/\s/g, '').toUpperCase() === sdkName.toUpperCase()) {
+                version = splitArray[1];
+            }
+        }
+        else if(result && result.Project && result.Project.Sdk
+            && result.Project.Sdk[0].$ && result.Project.Sdk[0].$.Name && result.Project.Sdk[0].$.Version) {
+                referenceType = sdkReference.Element;
+                if (result.Project.Sdk[0].$.Name.toUpperCase() === sdkName.toUpperCase()) {
```

Comment:
There could be spaces within element too, so trimming should also be done in this branch

Reply:
Technically it should not work if element has spaces, msbuild would throw an error that could not find the sdk. I think we should keep that way rather than fix it for them.

Reply:
You sure? I think most of it will be trimmed by the XML parser, usually. For example:
```xml
<Name>
   MySdk
</Name>
```
Has a lot of extra characters and is still valid and I think msbuild will take it no issues. Not sure if the parser we are using will trim newlines and indentation spaces...